runtime/commonMain/src/kotlinx/serialization/internal/Tuples.kt

```diff
@@ -63,18 +63,12 @@ public class MapEntrySerializer<K, V>(
     keySerializer: KSerializer<K>,
     valueSerializer: KSerializer<V>
 ) : KeyValueSerializer<K, V, Map.Entry<K, V>>(keySerializer, valueSerializer) {
-    private inner class MapEntryClassDesc : SerialClassDescImpl("kotlin.collections.Map.Entry") {
-        override val kind = StructureKind.MAP
-
-        init {
-            addElement("key")
-            pushDescriptor(keySerializer.descriptor)
-            addElement("value")
-            pushDescriptor(valueSerializer.descriptor)
-        }
+    // TODO map kind is most likely to be incorrect
```

Comment:
It allows to write it in JSON literally as map entry `"key":"value"`. It is likely used in protobuf, too